platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+
+![img_1.svg](img_1.svg)
+
+Also, we will extract a `PublicShare` for each `ShareId`: `sidᵢ` in the directory from the list of valid messages.
+The PublicShare for share `s` is computed by evaluating each polynomial commitment in the common set of messages at `sidᵢ` and then aggregating the results.
+
+At this point, the participant executing the scheme can start signing, sharing signatures, and validating individual signatures produced by other parties in the scheme.
+
+
+
+#####  Rekey Stage
+This Section describes the rekey process executed each time some of the scheme’s parameters change, such as the number of participants or the number of shares assigned to each participant.
+This rekeying process may also be conducted for reasons other than parameter changes; to rotate shares, for example.
+
+The rekeying process is similar to the bootstrap process, but it starts with the previous list of owned private `SecretShare`.
+The main difference with the genesis stage is that every participant generates a `TssMessage` out of each previously owned `SecretShare`.
+
+
+![img_4.svg](img_4.svg)
+
+Once finished, the list of `SecretShare`s will be updated but the previously generated aggregate public key remains the same.
+
+
+#####  Sign Stage
+After genesis or rekeying stages, the library can sign any message.
+Using each `SecretShare` owned by the participant, a message can be signed, producing a `TssSignature`
+
+Multiple signatures can be aggregated to create an aggregate `TssSignature`. An aggregate `TssSignature` can be validated against the LedgerId (public key)  if
+`t` (t=threshold) valid signatures are aggregated. If the threshold is met and the signature is valid, the library will respond with true; if not, it will respond with false.
+
+
+#### Security Considerations
+As long as an adversary Participant knows fewer than a threshold number of decryption keys, they cannot recover enough information to start forging threshold signatures.
+Adversarial participants may learn the shares of the participants whose keys they have compromised, but more is needed to recover the secret.
+For security, adversarial parties might choose low or zero entropy values for protocol inputs such as shareIds, but here will still be sufficient entropy in the overall protocol from honest nodes.
+
+#### Summary diagram
+The following diagram exposes the steps necessary for bootstrapping and using the library. In orange are all the steps where the library is involved.
+The rest of the steps need to happen but are outside the scope of the described library.
+
+![img_2.png](img_2.svg)
+
+
+### Architecture
+To implement the functionality detailed in the previous section, the following code structure is proposed:
+
+![img_5.png](img_5.svg)
+
+1. **TSS Lib**: The consensus node will use the TSS library to create shares, create TSS messages to send to other nodes,
+   assemble shared public keys (ledgerId), and sign the block node Merkle tree hash.
```

Comment:
Immaterial change - i think its best to just say that the consensus node will sign messages. We don't need to specify which.
```suggestion
   assemble shared public keys (ledgerId), and sign messages.
```